synapse/tests/test_lib_agenda.py

```diff
@@ -829,6 +827,105 @@ async def test_agenda_mirror_realtime(self):
                 data = stream.read()
                 self.isin("_Appt.edits() Invalid attribute received: invalid = 'newp'", data)
 
+    async def test_agenda_promotions(self):
+        async with self.getTestAha() as aha:
+
+            conf00 = {
+                'aha:provision': await aha.addAhaSvcProv('00.cortex')
+            }
+
+            async with self.getTestCore(conf=conf00) as core00:
+                self.false(core00.conf.get('mirror'))
+
+                msgs = await core00.stormlist('[it:dev:str=foo]')
+                self.stormHasNoWarnErr(msgs)
+
+                await core00.callStorm('cron.add --minute +1 { $lib.time.sleep(90) }')
+
+                prov01 = {'mirror': '00.cortex'}
+                conf01 = {
+                    'aha:provision': await aha.addAhaSvcProv('01.cortex', provinfo=prov01),
+                }
+
+                async with self.getTestCore(conf=conf01) as core01:
+                    msgs = await core01.stormlist('[it:dev:str=bar]')
+                    self.stormHasNoWarnErr(msgs)
+
+                    # Advance the ticks so the cronjob starts sooner
+                    core00.agenda._addTickOff(55)
+
+                    mesgs = []
+                    async for mesg in core00.behold():
+                        mesgs.append(mesg)
+                        break
+
+                    self.eq(mesgs[0]['event'], 'cron:start')
+
+                    # Inspect crons and tasks
+                    crons00 = await core00.callStorm('return($lib.cron.list())')
+                    self.len(1, crons00)
+                    # isrunning is synced via nexus so it should be true for both cortexes
+                    self.true(crons00[0].get('isrunning'))
+
+                    crons01 = await core01.callStorm('return($lib.cron.list())')
+                    self.len(1, crons01)
+                    # isrunning is synced via nexus so it should be true for both cortexes
+                    self.true(crons01[0].get('isrunning'))
+
+                    tasks00 = await core00.callStorm('return($lib.ps.list())')
+                    # Two tasks: one for the cronjob and one for the cronjob instance
```

Comment:
Looks like the two tasks are actually the cronjob and the `[it:dev:str=foo]` storm query that was previously run, those are apparently not being cleaned up?

Reply:
Is the `[it:dev:str=foo]` task the maintask being promoted?

Reply:
yep. this just sticks around forever?

Reply:
It stick around since that task can't be cancelled ( readily ). This is an artifact of view.storm promoting the current task; which is normally coming in as it's own task as a telepath api call.